network/src/network.rs

```diff
@@ -470,7 +498,17 @@ impl Network {
             Some(private_key) => private_key?,
             None => return Err(ErrorKind::Other("secret_key not set".to_owned()).into()),
         };
-        let listened_addresses = config.public_addresses.clone();
+        let listened_addresses = {
```

Comment:
HashMap implements  `FromIterator` trait, one line solution can be used here:
```suggestion
let listened_addresses: FnvHashMap<Multiaddr, u8> = config.public_addresses
    .iter().map(|addr| (addr.to_owned(), std::u8::MAX) ).collect();
```